lib/stat_tracker.rb

```diff
@@ -94,10 +94,24 @@ def average_goals_per_game #issue #8
   end
 
   def average_goals_by_season #issue #9
+    my_hash = Hash.new { |h,k| h[k.to_s] = [] }
+    count_of_games_by_season.each do |season, game_count|
+      my_hash[season] = [] 
+    end
 
-  end
-
+      count_of_games_by_season.each do |season, game_count|
+        game_sum_calc = []
+        games.each do |row| 
+          game_sum_calc << (row[:away_goals] + row[:home_goals]) if row[:season] == season
```

Comment:
Would it work to do the `each` loop on line #97 within the `each` loop starting on line 102? That way it is only looping through the `count_of_games_by_season` once rather than 2 separate times.